packages/filebrowser/src/listing.ts

```diff
@@ -936,6 +940,18 @@ export class DirListing extends Widget {
     );
   }
 
+  /**
+   * Update the setting to sort notebooks above files.
+   * This sorts the items again if the internal value is modified.
+   */
+  setNotebooksFirstSorting(isEnabled: boolean) {
+    let previousValue = this._sortNotebooksFirst;
+    this._sortNotebooksFirst = isEnabled;
+    if (this._sortNotebooksFirst !== previousValue) {
+      this.sort(this._sortState);
+    }
+  }
+
```

Comment:
Would you mind switching to a getter-setter? This is useful to have a getter when a setter is defined. So that external code can check what is the current value.

FYI: I also switched the `let` to `const` as it is a good practice to preferred `const` when a variable is expected to not be modified.

```suggestion
  /**
   * Update the setting to sort notebooks above files.
   * This sorts the items again if the internal value is modified.
   */
  get notebooksFirstSorting(): boolean {
    return this._sortNotebooksFirst;
  }
  set notebooksFirstSorting(isEnabled: boolean) {
    const previousValue = this._sortNotebooksFirst;
    this._sortNotebooksFirst = isEnabled;
    if (this._sortNotebooksFirst !== previousValue) {
      this.sort(this._sortState);
    }
  }

```

Reply:
I've used a method as I've copied the pattern from the column visibility. I do prefer the setter/getter though, thanks for the suggestion!

https://github.com/jupyterlab/jupyterlab/blob/d47d50bbeb2db80df09489eb7c9588c29a140d22/packages/filebrowser/src/listing.ts#L925C3-L941